test/integration/jenkinsnode/cistack/msp430f5529.py

```
@@ -0,0 +1,77 @@
+import os
+import sys
+import magic
+import re
+import subprocess
+from time import sleep
+import RPi.GPIO as GPIO
+from target import Target
+from pin import Pin
+
+errstr = "*** ERROR (msp430 target) "
+
+class MSP430(Target):
+    def __init__(self):
+        self.board = "msp430f5529-gcc"
+        self.arch = "msp430"
+        self.cpu = "msp430f5529"
+        self.pins = {
+            'rst' : Pin(name = 'rst', number = 17),
+            'pwr' : Pin(name = 'pwr', number = 27),
+            'opt' : Pin(name = 'opt', number = 22)  # optional
+        }
+    
+    def flash(self, binobj):
+        from utils import findBin
+        """Flash a binary file through USB connection on an MSP430 Launchpad"""
+
+        if not self.sanitycheck(binobj):
+            sys.exit("Binary file didn't pass a sanity check. Exiting.")
```

I know we are found of using `sys.exit` to fail fast but it may be worth considering removing these exits and returning back errors. If we bail with an exit there is no chance to cleanup or reset things to a nice state (reset board, shut down board, free pins).